service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/PlacementController.java

```diff
@@ -119,9 +121,20 @@ public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
         Placement currentPlacement = placementService.getPlacement(caseData, child);
         Placement previousPlacement = placementService.getPlacement(caseDataBefore, child);
 
+        // TODO refactor: this logic is confusing. Suggested: new method isNewOrUpdated...
         if (!isUpdatingExistingPlacement(previousPlacement, currentPlacement)) {
             publishPlacementApplicationUploadEvent(callbackRequest);
         }
+
+        List<String> currentDocumentUrls = getBinaryUrlsForNoticeOfPlacementOrder(caseData.getPlacements());
+        List<String> previousDocumentUrls = getBinaryUrlsForNoticeOfPlacementOrder(caseDataBefore.getPlacements());
+        currentDocumentUrls.removeAll(previousDocumentUrls);
```

Comment:
A start for splitting out the logic would probably be to move this to the service

Reply:
I have split to private methods, I am currently unsure where these should exist outside of the controller